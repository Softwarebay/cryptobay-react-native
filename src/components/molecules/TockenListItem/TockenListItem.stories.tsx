import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import StorybookView from '../../../../storybook/StorybookView';
import { TockenListItem } from './TockenListItem';

storiesOf('Molecules/TockenListItem', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Item', () => (
    <TockenListItem
      tockenPicture={0}
      tockenName={text('tockenName', 'Binance Coin')}
      tockenDolarValue={text('tockenDolarVale', '$226.69')}
      tockenPercentValueChange={number('tockenPercentValueChange', 2)}
      tockenValue={text('tockenValue', '19.2371 BNB')}
    />
  ));
