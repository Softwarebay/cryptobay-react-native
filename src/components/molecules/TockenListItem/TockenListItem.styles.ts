import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  tockenListItem: {
    width: '100%',
    marginVertical: 8,
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  tockenPicture: {
    width: 34,
    height: 34,
    backgroundColor: '#44485F',
    borderRadius: 34,
  },
  tockenDataWrapper: {
    width: '60%',
  },
  tockenCurrencyDataWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  tockenDolarValue: {
    marginRight: 8,
  },
  tockenPriceWrapper: {
    height: '100%',
    justifyContent: 'flex-start',
  },
});
