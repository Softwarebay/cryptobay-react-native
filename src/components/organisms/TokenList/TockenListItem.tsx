import React from 'react';
import { View } from 'react-native';

import { Token } from '../../../types/token.type';
import { Badge, Caption, Label, Picture } from '../../atoms';

import { tockenListItemStyles } from './TokenList.styles';

export const TokenListItem: React.FC<Token> = ({
  picture,
  name,
  shortName,
  price,
  percentagePoint,
  value,
}: Token) => {
  const captionLabel = `$${price}`;
  const percentageLabel =
    percentagePoint > 0 ? `+${percentagePoint}%` : `${percentagePoint}%`;
  const variant = percentagePoint > 0 ? 'green' : 'red';
  const valueLabel = `${value} ${shortName}`;

  return (
    <View style={tockenListItemStyles.container}>
      <View style={tockenListItemStyles.dataWrapper}>
        <Picture style={tockenListItemStyles.picture} pictureFile={picture} />

        <View>
          <Label value={name} />

          <View style={tockenListItemStyles.valueData}>
            <Caption
              style={tockenListItemStyles.caption}
              value={captionLabel}
            />
            <Badge value={percentageLabel} variant={variant} />
          </View>
        </View>
      </View>

      <Label value={valueLabel} />
    </View>
  );
};
