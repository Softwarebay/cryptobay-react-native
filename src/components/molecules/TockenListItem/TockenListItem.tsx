import { ImageSourcePropType, View } from 'react-native';
import React from 'react';
import { Badge, Caption, Label, Picture } from '../../atoms';
import { styles } from './TockenListItem.styles';

type TockenListItemProps = {
  tockenPicture: ImageSourcePropType;
  tockenName: string;
  tockenDolarValue: string;
  tockenPercentValueChange: number;
  tockenValue: string;
};

export const TockenListItem: React.FC<TockenListItemProps> = ({
  tockenPicture,
  tockenName,
  tockenDolarValue,
  tockenPercentValueChange,
  tockenValue,
}) => {
  return (
    <View style={styles.tockenListItem}>
      <Picture pictureFile={tockenPicture} style={styles.tockenPicture} />
      <View style={styles.tockenDataWrapper}>
        <Label value={tockenName} />

        <View style={styles.tockenCurrencyDataWrapper}>
          <Caption value={tockenDolarValue} style={styles.tockenDolarValue} />
          <Badge value={tockenPercentValueChange} />
        </View>
      </View>

      <View style={styles.tockenPriceWrapper}>
        <Label value={tockenValue} />
      </View>
    </View>
  );
};
