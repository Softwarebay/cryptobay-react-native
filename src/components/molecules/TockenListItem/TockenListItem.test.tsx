import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { TockenListItem } from './TockenListItem';

describe('TockenListItem Molecule', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <TockenListItem
          tockenPicture={0}
          tockenName={''}
          tockenDolarValue={''}
          tockenPercentValueChange={0}
          tockenValue={''}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
