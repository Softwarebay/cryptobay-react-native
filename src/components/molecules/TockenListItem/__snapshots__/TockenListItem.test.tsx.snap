// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TockenListItem Molecule Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 8,
      "padding": 8,
      "width": "100%",
    }
  }
>
  <Image
    source={0}
    style={
      Object {
        "backgroundColor": "#44485F",
        "borderRadius": 34,
        "height": 34,
        "width": 34,
      }
    }
  />
  <View
    style={
      Object {
        "width": "60%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 18,
            },
            Object {
              "color": "#ABAFC4",
            },
            Object {
              "marginRight": 8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#76E268",
            },
            undefined,
          ]
        }
      >
        +
        0
        %
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    />
  </View>
</View>
`;
